TreeDB/documentservice/service_benchmark_cache_test.go

```
@@ -84,6 +84,51 @@ func TestServiceBenchmarkVectorSearchCacheWarmOnOptimizeAndReuse(t *testing.T) {
+	if _, err := svc.UpsertDocuments(ctx, "raw_dense_cache", UpsertDocumentsRequest{Documents: []Document{
+		{ID: "a", Embedding: []float32{1, 0}, Meta: map[string]any{"user_id": "alpha"}},
+		{ID: "b", Embedding: []float32{0, 1}, Meta: map[string]any{"user_id": "beta"}},
+	}}); err != nil {
+		t.Fatalf("UpsertDocuments: %v", err)
+	}
+	if _, err := svc.OptimizeIndex(ctx, "raw_dense_cache", OptimizeIndexRequest{}); err != nil {
+		t.Fatalf("OptimizeIndex: %v", err)
+	}
+
+	reader := New(collections.NewCollectionManager(db))
+	defer reader.Close()
```

golangci-lint (2.13.2)

[error] Error return value of `reader.Close` is not checked

(errcheck)